lib/Experian/IDAuth.pm

```diff
@@ -299,13 +284,9 @@ sub _xml_as_hash {
 sub _get_result_proveid {
     my $self = shift;
 
-    my $report = $self->{result_as_xml} || die 'needs xml report';
+    my $report = $self->{result_as_xml} or die 'needs xml report';
 
-    my $twig = eval { XML::Twig->parse($report) } || do {
-        my $err = $@;
-        warn "could not parse xml report: $err";
-        return;
-    };
+    my $twig = eval { XML::Twig->parse($report) } or die "could not parse xml report: $@";
```

Comment:
Seems like a `Try::Tiny` could be useful here, as there's one too many `eval BLOCK`s.